go/enclave/rollupchain/rollup_chain.go

```diff
@@ -144,21 +148,27 @@ func (rc *RollupChain) ProcessL1Block(block types.Block, isLatest bool) (*common
 	}
 
 	// We update the L1 and L2 chain heads.
-	l2Head, isUpdatedRollupHead, err := rc.updateHeads(&block)
+	oldL2Head, err := rc.storage.FetchHeadRollupForL1Block(block.ParentHash())
```

Comment:
Instead of returning `isUpdatedRollupHead` from the depths of `updateHeads`, I just calculate whether it's changed here, to make the flow easier to follow.